packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,186 @@
+import type { ILinkQueue, ILink } from '@comunica/bus-rdf-resolve-hypermedia-links-queue';
+import { LinkQueueWrapper } from '@comunica/bus-rdf-resolve-hypermedia-links-queue';
+import type { Logger } from '@comunica/types';
+
+/**
+ * A link queue that log information about event happening in the queue
+ */
+export class LinkQueueLogger extends LinkQueueWrapper {
+  public readonly query: string;
+  private readonly logger: Logger;
+  private readonly linkProductionRatio: ILinkProductionActorRatio = {
+    pushEvent: {},
+    popEvent: {},
+  };
+
+  public static readonly LINK_QUEUE_EVENT_NAME = '<Link queue occupancy>';
```

Comment:
```suggestion
  public static readonly LINK_QUEUE_EVENT_NAME = 'Link queue changed';
```

Can probably also be inlined after removing the empty-check.